arangod/IResearch/IResearchAqlAnalyzer.cpp

```diff
@@ -652,11 +722,28 @@ bool AqlAnalyzer::reset(irs::string_ref const& field) noexcept {
       optimizer.createPlans(std::move(plan), _query->queryOptions(), false);
 
       _plan = optimizer.stealBest();
+
+      // try to optimize
+      if(optimize(field)) {
+        _optimizedField.assign(field.c_str(), field.size());
+        return true;
+      }
+
     } else {
+      // plan exists
+
       for (auto node : _bindedNodes) {
         node->setStringValue(field.c_str(), field.size());
       }
       _engine.reset();
+
+      if(_optimizedField != field) {
+        // try to optimize
+        if(optimize(field)) {
```

Comment:
Why we are running optimize again? If we got here - plan was already inspected. We know for sure  if we should optimize or not.